### DISCUSSION TITLE
Need Github actions for Dotnet framework unit testing

### DISCUSSION THREAD

**Discussion by sujayraibagkar** [community/community#4726] [Actions]
Hi ,
We are migrating from Azure DevOps to GitHub Actions. We are trying to build and test Dotnet framework application.
We did not find any action to test the Dotnet framework applications and we have some restrictions on using 3rd party actions from marketplace.
We would really like to have github actions for testing similar to one in Azure DevOps. https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/test/vstest?view=azure-devops 
This will enable us to use verified action from GitHub in our code base.
https://github.com/microsoft/vstest

**Comment by ccolella-mdc** [comment]
you can use the `run` keyword in a step to run a command line script. you can use `dotnet test` or `vstest.console.exe` (on windows) via a script to run tests. There is also [this action](https://github.com/actions/setup-dotnet) to setup the `dotnet` command

**Comment by ethomson** [comment]
Hi @sujayraibagkar - I'm afraid that we don't have the capabilities to build all the actions ourselves.  I will relay your request to some friends in the Microsoft developer tools space to see if this is something that they can help with.  Thanks for the feedback!
